segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/IndexWriter.java

```diff
@@ -318,4 +340,37 @@ private AttributeUpdate generateBackpointerRemoval(long fromOffset) {
     }
 
     //endregion
+
+    //region Helper Classes
+
+    private static class UpdateInstructions {
```

Comment:
Is there any possibility of concurrency when updating these values? I'm wondering if there is any need for a mechanism to avoid races or inconsistencies.

Reply:
IndexWriter executes on a single thread; there is no concurrency inside it.